File: java/src/SearchReplaceResponder.java

class SearchReplaceResponder extends ResultResponder {
	public String title() {
		return "Search/Replace: " + search_text() + "/" + replace_text();
	}

	public boolean traverse(WikiPage page) {
		if (page.text().contains(search_text())) {
			page.setText(page.text().replaceAll(search_text(), replace_text()));
			return true;
		}
		return false;
	}

	private String search_text() {
		return request.data().get("search_text");
	}

	private String replace_text() {
		return request.data().get("replace");
	}
}

File: csharp/SearchReplaceResponder.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;

namespace WikiSearch
{
	public class SearchReplaceResponder : ResultResponder
	{
		public override string Title()
		{
			return "Search/Replace: " + SearchText() + "/" + ReplaceText();
		}

		public override bool Traverse(WikiPage page)
		{
			if (page.Text.Contains(SearchText())) {
				page.SetText(page.Text.Replace(SearchText(), ReplaceText()));
				return true;
			}
			return false;
		}

		private string SearchText()
		{
			return request.Data["search_text"];
		}

		private string ReplaceText()
		{
			return request.Data["replace"];
		}
	}
}

